pulsar-client/src/main/java/org/apache/pulsar/client/api/ClientBuilder.java

```diff
@@ -0,0 +1,265 @@
+     *             failed to instantiate specified Authentication-Plugin
+     */
+    ClientBuilder authentication(String authPluginClassName, Map<String, String> authParams)
+            throws UnsupportedAuthenticationException;
+
+    /**
+     * Set the operation timeout <i>(default: 30 seconds)</i>
+     * <p>
+     * Producer-create, subscribe and unsubscribe operations will be retried until this interval, after which the
+     * operation will be maked as failed
+     *
+     * @param operationTimeout
+     *            operation timeout
+     * @param unit
+     *            time unit for {@code operationTimeout}
+     */
+    ClientBuilder operationTimeout(int operationTimeout, TimeUnit unit);
+
+    /**
+     * Set the number of threads to be used for handling connections to brokers <i>(default: 1 thread)</i>
+     *
+     * @param numIoThreads
+     */
+    ClientBuilder ioThreads(int numIoThreads);
+
+    /**
+     * Set the number of threads to be used for message listeners <i>(default: 1 thread)</i>
+     *
+     * @param numListenerThreads
+     */
+    ClientBuilder listenerThreads(int numListenerThreads);
+
+    /**
+     * Sets the max number of connection that the client library will open to a single broker.
+     * <p>
+     * By default, the connection pool will use a single connection for all the producers and consumers. Increasing this
+     * parameter may improve throughput when using many producers over a high latency connection.
+     * <p>
+     *
+     * @param connectionsPerBroker
+     *            max number of connections per broker (needs to be greater than 0)
+     */
+    ClientBuilder connectionsPerBroker(int connectionsPerBroker);
+
+    /**
+     * Configure whether to use TCP no-delay flag on the connection, to disable Nagle algorithm.
+     * <p>
+     * No-delay features make sure packets are sent out on the network as soon as possible, and it's critical to achieve
+     * low latency publishes. On the other hand, sending out a huge number of small packets might limit the overall
+     * throughput, so if latency is not a concern, it's advisable to set the <code>useTcpNoDelay</code> flag to false.
+     * <p>
+     * Default value is true
+     *
+     * @param enableTcpNoDelay
+     */
+    ClientBuilder enableTcpNoDelay(boolean enableTcpNoDelay);
+
+    /**
+     * Configure whether to use TLS encryption on the connection <i>(default: false)</i>
+     *
+     * @param enableTLS
+     */
+    ClientBuilder enableTLS(boolean enableTLS);
+
+    /**
+     * Set the path to the trusted TLS certificate file
+     *
+     * @param tlsTrustCertsFilePath
+     */
+    ClientBuilder tlsTrustCertsFilePath(String tlsTrustCertsFilePath);
+
+    /**
+     * Configure whether the Pulsar client accept untrusted TLS certificate from broker <i>(default: false)</i>
+     *
+     * @param tlsAllowInsecureConnection
+     */
+    ClientBuilder tlsAllowInsecureConnection(boolean tlsAllowInsecureConnection);
```

Comment:
How about "allowTlsInsecureConnection"?

Reply:
Sure, with the builder it makes sense a more "imperative" style